### DISCUSSION TITLE
Using the Instant Prompt without `p10k`

### DISCUSSION THREAD

**Discussion by aboqasem** [romkatv/powerlevel10k#2607] [General]
May I know whether Instant Prompt, could be used independently from `p10k`? I stumbled upon [a reddit post by @romkatv](https://www.reddit.com/r/zsh/comments/dht4zt/make_zsh_start_instantly_with_this_one_weird_trick/) and was wondering if it achieves the same behaviour.

**Comment by romkatv** [comment]
See https://github.com/starship/starship/issues/888#issuecomment-580127661 for details on the implementation of instant prompt.

May I ask why you want to use Instant Prompt without `p10k`?

**Reply by aboqasem** [reply]
Honestly, I do not fully understand what `p10k` does. From what I understood, it's a prompt theme, but I am using Starship, so I am not sure if I can install `p10k` 😅

**Comment by romkatv** [comment]
Powerlevel10k does the same thing as Starship -- it sets your prompt. The advantage of Starship is that you can use it with Bash and Fish (Powerlevel10k is only for Zsh). The advantage of Powerlevel10k is that it's much faster and has more features. Instant Prompt is one of the features that Powerlevel10k has but Starship does not.

**Reply by aboqasem** [reply]
Thanks a lot for the clarification!

I spent 3 hours today reading about Starship and [configuring it](https://github.com/aboqasem/dotfiles/commit/aaf793960c096a7800ee3385fa8abcc7a38a516d#diff-f055ddb9b3ffac44f89ccab1c855b6e350ff908e45e1b1980e2613487da0f716), will look into switching to Powerlevel10k probably next weekend.

**Comment by romkatv** [comment]
3 hours?! Oh my.

It takes 3 minutes to configure powerlevel10k. You can try it and switch back to Starship if you don't like the results. Here's one way to do it:

```zsh
git clone --depth=1 https://github.com/romkatv/powerlevel10k.git ~/powerlevel10k
echo 'source ~/powerlevel10k/powerlevel10k.zsh-theme' >>~/.zshrc
exec zsh
```

This will start an interactive configuration wizard that you can use to configure your prompt.

If you don't like the result, delete the last two lines from `~/.zshrc` and you are back to Starship.

**Reply by aboqasem** [reply]
Haha yeah, not exactly but on and off, also I usually read the whole documentation before using the tool it took a while to finish it 😄.

Sure will try it some time!

Again, thanks a lot!

**Comment by romkatv** [comment]
You are in luck! Powerlevel10k does not have documentation in a conventional sense.

On a more serious note, I should've mentioned this as an obvious downside of Powerlevel10k compared to Starship. Nevertheless, I encourage you to give it a try.

**Reply by aboqasem** [reply]
Thanks! Giving it a try now: https://github.com/aboqasem/dotfiles/pull/3.

I would say the speed is awesome; I'm used to `Enter + Cmd-K` to clear the terminal but keep the prompt (since the prompt is two lines), with powerlevel10k I cannot do that because `Enter` immediately shows the prompt haha!

Need to explore that huge config file, want to disable the machine and directory logos and enable other tool logos, and also configure colors.

May I know if there's a solution to this: https://github.com/romkatv/powerlevel10k/issues/2467? Is it putting `omz update` before the `# Enable Powerlevel10k instant prompt.`?

**Reply by romkatv** [reply]
> I would say the speed is awesome; I'm used to `Enter + Cmd-K` to clear the terminal but keep the prompt (since the prompt is two lines), with powerlevel10k I cannot do that because `Enter` immediately shows the prompt haha!

This is awesome 😄 I've put a lot of effort into performance. If you are curious about the methodology, tools and benchmarks, check out https://github.com/romkatv/zsh-bench.

As far as logos go, I recommend choosing "few icons" rather than "many  icons" in `p10k configure`. You'll still get the informative icons but the decorative icons (like the OS logo and directory icons) will disappear. There is also https://github.com/romkatv/powerlevel10k?tab=readme-ov-file#what-is-the-best-prompt-style-in-the-configuration-wizard

I haven't yet looked into the omz update issue. Even when I do get around to it, I would strongly advise disabling auto-update on zsh startup anyway. Since you have a dotfiles repository, make omz a subrepo or a subtree. This way you can version control your dependencies, analyze diffs when something breaks, revert, etc. While at it, turn into submodules zsh-syntax-highlighting, powerlevel10k and other third party repos you use.
